src/cargo/ops/cargo_output_depinfo.rs

```diff
@@ -0,0 +1,148 @@
+    let mut result = Vec::new();
+    for line in dep_txt.lines() {
+        let mut line_spl = line.split(": ");
+        if let Some(target) = line_spl.next() {
+            if let Some(deps) = line_spl.next() {
+                let deps = deps.split_whitespace().map(ToOwned::to_owned).collect();
+                result.push(DepLine {
+                    target: target.to_string(),
+                    deps: deps,
+                });
+            }
+        }
+    }
+    Ok(DepFile {
+        dir: dir.to_string(),
+        deps: result,
+    })
+}
+
+fn add_deps(depfile: &DepFile, deps: &mut HashSet<PathBuf>) {
+    let dep_dir = PathBuf::from(&depfile.dir);
+    for depline in &depfile.deps {
+        for dep in &depline.deps {
+            deps.insert(dep_dir.join(dep));
+        }
+    }
+}
+
+// TODO: this should be based on CompileFilter
+fn match_target(target: &Target, example: Option<&str>, is_lib: bool)
+    -> Option<(&'static str, bool)>
+{
+    match *target.kind() {
+        TargetKind::Lib(..) => Some(("lib", is_lib)),
+        TargetKind::Bin => {
+            if example.is_none() && !is_lib {
+                Some(("bin", true))
+            } else {
+                None
+            }
+        }
+        TargetKind::Example => {
+            if let Some(e) = example {
+                if e == target.name() {
+                    return Some(("example", true));
+                }
+            }
+            None
+        }
+        _ => None
+    }
+}
+
+fn target_filename(context: &mut Context, unit: &Unit) -> CargoResult<PathBuf> {
+    let (dir, base) = context.link_stem(&unit).ok_or(internal("can't get link stem"))?;
+    if unit.target.is_lib() {
+        Ok(dir.join(["lib", &base, ".rlib"].concat()))
+    } else {
+        Ok(dir.join(base))
+    }
+}
+
+pub fn output_depinfo(path: &str, ws: &Workspace, config: &Config,
+                      build_config: BuildConfig) -> CargoResult<()> {
+    let (packages, resolve) = ops::resolve_ws(&ws)?;
+    let profiles = ws.profiles();
+    let profile = if build_config.release {
+        &profiles.release
+    } else {
+        &profiles.dev
+    };
+    let mut context = Context::new(&ws, &resolve, &packages, &config, build_config, &profiles)?;
+    let basedir = None;  // TODO: should this be a command line option? Maybe always cwd?
+    let example = None;  // TODO: use CompileFilter from config
+    let mut target_fn = String::new();
+    for package in ws.members() {
+        let mut deps = HashSet::new();
+        for target in package.targets() {
+            if let Some((kind, is_main)) = match_target(target, example, false) {
+                let unit = Unit {
+                    pkg: package,
+                    target: target,
+                    profile: &profile,
+                    kind: Kind::Target,
+                };
+                let filename = ["dep-", kind, "-", &context.file_stem(&unit)].concat();
```

Comment:
Here I think we may want to tweak how fingerprints are managed slightly. Could we maintain a map of fingerprints for units (e.g. active paths) on the `Context` and then we scrape that list afterwards? That way we don't have to re-parse anything or re-guess filenames and such.